client_overlay/src/tabcontrollers/WalkInPlaceTabController.cpp

```diff
@@ -455,111 +455,29 @@ namespace walkinplace {
-		double devRoll = (180 * std::asin(devRight.v[1])) / M_PI;
+		vr::HmdVector3_t cont1Vel = { 0, 0, 0 };
+		vr::HmdVector3_t cont2Vel = { 0, 0, 0 };
+		vr::HmdVector3_t tracker1Vel = { 0, 0, 0 };
+		vr::HmdVector3_t tracker2Vel = { 0, 0, 0 };
 
+		vr::HmdVector3d_t hmdRotVel_ = getHMDRotVel(tdiff);
 		if (dataModel.n_cols > 5) {
-			hmdRotVel.v[0] = (devYaw - lastHmdRot.v[0]) / tdiff;
-			hmdRotVel.v[1] = (devPitch - lastHmdRot.v[1]) / tdiff;
-			hmdRotVel.v[2] = (devRoll - lastHmdRot.v[2]) / tdiff;
+			hmdRotVel = hmdRotVel_;
 		}
 
-		lastHmdRot.v[0] = devYaw;
-		lastHmdRot.v[1] = devPitch;
-		lastHmdRot.v[2] = devRoll;
-
 		if (controller1ID != vr::k_unTrackedDeviceIndexInvalid) {
-			cont1Vel.v[0] = latestDevicePoses[controller1ID].vVelocity.v[0];
-			cont1Vel.v[1] = latestDevicePoses[controller1ID].vVelocity.v[1];
-			cont1Vel.v[2] = latestDevicePoses[controller1ID].vVelocity.v[2];
-			qRotation = vrmath::quaternionFromRotationMatrix(latestDevicePoses[controller1ID].mDeviceToAbsoluteTracking);
-			devForward = vrmath::quaternionRotateVector(qRotation, forward);
-			devRight = vrmath::quaternionRotateVector(qRotation, right);
-
-			//cont1Vel = vrmath::quaternionRotateVector(qRotation, cont1Vel);
-
-			devYaw = (180 * std::asin(devForward.v[0])) / M_PI;
-			devPitch = (180 * std::asin(devForward.v[1])) / M_PI;
-			devRoll = (180 * std::asin(devRight.v[1])) / M_PI;
-
+			cont1Vel = latestDevicePoses[controller1ID].vVelocity;
 			if (controller2ID != vr::k_unTrackedDeviceIndexInvalid) {
-				cont2Vel.v[0] = latestDevicePoses[controller2ID].vVelocity.v[0];
-				cont2Vel.v[1] = latestDevicePoses[controller2ID].vVelocity.v[1];
-				cont2Vel.v[2] = latestDevicePoses[controller2ID].vVelocity.v[2];
-				qRotation = vrmath::quaternionFromRotationMatrix(latestDevicePoses[controller2ID].mDeviceToAbsoluteTracking);
-				devForward = vrmath::quaternionRotateVector(qRotation, forward);
-				devRight = vrmath::quaternionRotateVector(qRotation, right);
-
-				//cont2Vel = vrmath::quaternionRotateVector(qRotation, cont2Vel);
-
-				devYaw = (180 * std::asin(devForward.v[0])) / M_PI;
-				devPitch = (180 * std::asin(devForward.v[1])) / M_PI;
-				devRoll = (180 * std::asin(devRight.v[1])) / M_PI;
+				cont2Vel = latestDevicePoses[controller2ID].vVelocity;
 			}
 		}
 
 		if (tracker1ID != vr::k_unTrackedDeviceIndexInvalid) {
-			tracker1Vel.v[0] = latestDevicePoses[tracker1ID].vVelocity.v[0];
-			tracker1Vel.v[1] = latestDevicePoses[tracker1ID].vVelocity.v[1];
-			tracker1Vel.v[2] = latestDevicePoses[tracker1ID].vVelocity.v[2];
-			qRotation = vrmath::quaternionFromRotationMatrix(latestDevicePoses[tracker1ID].mDeviceToAbsoluteTracking);
-			devForward = vrmath::quaternionRotateVector(qRotation, forward);
-			devRight = vrmath::quaternionRotateVector(qRotation, right);
-
-			//tracker1Vel = vrmath::quaternionRotateVector(qRotation, tracker1Vel);
-
-			devYaw = (180 * std::asin(devForward.v[0])) / M_PI;
```

Comment:
I think these were part of my initial look into basing the movement direction off the trackers (say like mounted to a belt on the waist) looks like that is why they are never used. Just a note, could pick that back up later.